kafka/consumer.go

```diff
@@ -150,6 +151,24 @@ func (cfg *ConsumerConfig) finalize() error {
 	if cfg.FetchMinBytes < 0 {
 		errs = append(errs, errors.New("kafka: fetch min bytes cannot be negative"))
 	}
+	if cfg.BrokerMaxReadBytes < 0 {
+		errs = append(errs, errors.New("kafka: broker max read bytes cannot be negative"))
+	}
+	if cfg.BrokerMaxReadBytes > 1<<30 {
+		cfg.Logger.Info("kafka: BrokerMaxReadBytes exceeds 1GiB, setting to 1GiB")
+		cfg.BrokerMaxReadBytes = 1 << 30
+	}
+	if cfg.MaxPollBytes > 0 {
```

Comment:
Should we also validate that `BrokerMaxReadBytes ` is GTE than `MaxPollBytes` and set it to the bigger value otherwise?

Reply:
Isn't that what the code block above does?

Reply:
Not really, what I mean right now it's possible to set `BrokerMaxReadBytes=1` and `MaxPollBytes=1000000` and we will accept this configuration.

Reply:
ah good catch, added some more validation and tests.